docs/examples/module-temperature-check.py

```diff
@@ -0,0 +1,86 @@
+"""
+Module Temperature Check
+========================
+
+Test whether the module temperature is correlated with irradiance.
+"""
+
+# %%
+# Testing the correlation between module temperature and irradiance
+# measurements can help identify if there are issues with the module
+# temperature sensor.
+# In this example, we demonstrate how to use
+# :py:func:`pvanalytics.quality.weather.module_temperature_check`, which
+# runs a linear regression model of module temperature vs. irradiance. Model
+# performance is then assessed by the Pearson correlation coefficient.
+# If it meets a minimum threshold, function outputs a True boolean.
+# If not, it outputs a False boolean.
+
+import pvanalytics
+from pvanalytics.quality.weather import module_temperature_check
+import matplotlib.pyplot as plt
+import pandas as pd
+from scipy.stats import linregress
+import pathlib
+
+# %%
+# First, we read in example data from the NREL SERF West system, which contains
+# data for module temperature and irradiance under the 'module_temp_1__781'
+# and 'poa_irradiance__771' columns, respectively. This data set contains
+# 15-minute averaged measurements, and is available via the NREL PVDAQ database
+# as system 51.
+
+pvanalytics_dir = pathlib.Path(pvanalytics.__file__).parent
+serf_east_file = pvanalytics_dir / 'data' / 'serf_west_15min.csv'
+data = pd.read_csv(serf_east_file, index_col=0, parse_dates=True)
+print(data[['module_temp_1__781', 'poa_irradiance__771']].head(10))
+
+
+# %%
+# Plot the module temperature to visualize it.
+data['module_temp_1__781'].plot()
+plt.xlabel("Date")
+plt.ylabel("Module Temperature (deg C)")
+plt.xticks(rotation=25)
+plt.tight_layout()
+plt.show()
+
+
+# %%
+# Plot the POA irradiance to visualize it.
+data['poa_irradiance__771'].plot()
+plt.xlabel("Date")
+plt.ylabel("POA irradiance (W/m^2)")
+plt.xticks(rotation=25)
+plt.tight_layout()
+plt.show()
+
```

Comment:
I think we should filter out night-time periods before the regression. Including nighttime is likely affecting the correlation.

Reply:
Good catch @cwhanse. I used the day-night filter to take nighttime data out and it cleaned out all of the 0-irradiance values at the bottom of the graphic: 

Before: 
![image](https://user-images.githubusercontent.com/70228568/188193601-0ce03358-5aeb-4335-9183-9d2a6370f03f.png)

After: 
![image](https://user-images.githubusercontent.com/70228568/188193710-91c81398-93c6-4d10-81bd-93c9f8439552.png)

Pushed this new doc update in https://github.com/pvlib/pvanalytics/pull/151/commits/981ed69b3aadaa760592c25cc65a440b68dbf1e2